### ISSUE TITLE
Since Debian Testing/Unstable (between Debian 10-11), Debian 11 (2021) and Ubuntu based on it do not accept the latest FOG stable (1.5.9)

### ISSUE THREAD

**Issue by Neustradamus** [FOGProject/fogproject#542]
Dear @FOGProject team,

Currently latest FOG official release build (1.5.9) and master code do not work with Debian 11 (2021-08-14) and all Ubuntu versions based on Debian Testing/Unstable (between Debian 10 and 11)...

Important : Debian 12 will be released in some months (2023).

The problem exist since a very long time:
- Debian Testing/Unstable (development between Debian 10 and 11)
- Debian 11 "Bullseye" official release (2021-08-14), soon two years: https://www.debian.org/releases/bullseye/
- Always here for with the current Debian 12 Testing/Unstable

It is possible to have a new FOG stable release build to fix it?

For example a 1.5.10 version?

Thanks in advance.

Debian:
- https://packages.debian.org/search?keywords=php-gettext
- https://packages.debian.org/search?keywords=php-php-gettext

Ubuntu:
- https://packages.ubuntu.com/search?keywords=php-gettext
- https://packages.ubuntu.com/search?keywords=php-php-gettext

Linked to:
- https://github.com/FOGProject/fogproject/issues/429
- https://github.com/FOGProject/fogproject/issues/433
- https://github.com/FOGProject/fogproject/issues/448
- https://github.com/FOGProject/fogproject/issues/449
- https://github.com/FOGProject/fogproject/issues/451
- https://github.com/FOGProject/fogproject/issues/453
- https://github.com/FOGProject/fogproject/issues/484
- https://github.com/FOGProject/fogproject/issues/516
- https://github.com/FOGProject/fogproject/issues/525
- https://github.com/FOGProject/fogproject/issues/532
- https://forums.fogproject.org/topic/10850/removing-the-apache-and-php-packages-failed
- https://forums.fogproject.org/topic/13568/nfs-service-is-failing-to-run
- https://forums.fogproject.org/topic/14587/cannot-install-fog-1-5-8-on-ubuntu
- https://forums.fogproject.org/topic/14696/fog-installation-with-ubuntu-focal
- https://forums.fogproject.org/topic/15370/dev-branch-1-5-9-79-installing-package-php-gettext-failed-has-to-use-x-switch-to-get-it-to-work
- https://forums.fogproject.org/topic/15373/fog-installation-failed
- https://forums.fogproject.org/topic/15702/fog-installation-error-e-unable-to-locate-package-php-gettext
- https://forums.fogproject.org/topic/15914/install-fog-on-ubuntu-server-21-10-issues
- https://forums.fogproject.org/topic/16104/how-to-install-php-gettext
- https://forums.fogproject.org/topic/16105/ubunutu-server-21-10-install-fail
- https://forums.fogproject.org/topic/16107/debian-install-failed
- https://forums.fogproject.org/topic/16436/fog-install-php-failed
- https://forums.fogproject.org/topic/16478/1-5-9-failing-to-install-on-ubuntu
- https://forums.fogproject.org/topic/16541/fix-problem-with-php-gettext
- https://forums.fogproject.org/topic/16565/installing-1-5-9-on-debian-bullseye-fails
- https://forums.fogproject.org/topic/16632/fog-detected-a-potential-need-to-reinstall-php-and-apache-files
- ...

**Comment by mastacontrola** [comment]
This mostly seems to revolve around the php-gettext, which is now installed as a basic value of php-common from the repositories with Debian/Ubuntu.

Having to install php-php-gettext is useless from my experimentation.


The issue of 1.5.10 or any "stable" release is stable is a static term. So when we release a build of FOG, it's stable - yes, for the timing and technology (software, hardware, etc....) at that static point in time.

With software constantly changing and OS's constantly updating, stable is a term that is almost pointless to use regarding "forward" thinking.

That said, I think simply removing the requirement of installing the package named php-gettext will fix the issue as it is now installed as a common package on these Debian variant machines.

I'm sure there's more information than just the php-gettext issue, but this has been the "main" thing I've seen on the forums and is corrected in my testing by simply removing this from the packages line in /opt/fog/.fogsettings.

I don't know if @Sebastian-Roth or anybody else has added the php-php-gettext, but hopefully this will help others out in general.

**Comment by Sebastian-Roth** [comment]
@Neustradamus You are right, should really get out a new release. Keep repeating myself but this is only gonna happen when people actually have the time to do it. Get involved to make it go faster.

There will be a new release this spring but the more people help the sooner it'll be.

> The problem exist since a very long time...

Right and it is fixed in the dev-branch version since several months too - [reference](https://github.com/FOGProject/fogproject/commit/a19a09fcfd1857b7dbcbb1bccab7684ceb8fec64).

> Important : Debian 12 will be released in some months (2023).

### Yes, I take this as the main topic here.

@mastacontrola said:
> That said, I think simply removing the requirement of installing the package named php-gettext will fix the issue as it is now installed as a common package on these Debian variant machines.

Absolutely.

> I don't know if @Sebastian-Roth or anybody else has added the php-php-gettext, but hopefully this will help others out in general.

Don't think so ([1](https://github.com/FOGProject/fogproject/commit/88e241ac5367a5d406ee7bbaf92e064e51f13f44) and [2](https://github.com/FOGProject/fogproject/commit/1449923b4c975669c08827383e1d363441e9a78e)).

**Comment by Sebastian-Roth** [comment]
@Neustradamus Just tried to do a test install of FOG dev-branch on latest Debian testing. Seems like testing is still too buggy. I was able to manually fix a GRUB install setup issue [manually](https://askubuntu.com/a/1131970) but then on bootup I get the following messages:
```
/dev/sda1 has unsupported feature(s): FEATURE_C12
e2fsck: Get a newer version of e2fsck!
```
Have not found a solution to that and so testing dev-branch is not possible yet.

**Comment by Neustradamus** [comment]
@Sebastian-Roth: When I have tested Debian 11, I had same about GRUB but the problem is not linked to FOG.
It is linked to "Legacy" or "UEFI" mode, if I recall.

And the first step is to have a compatible version with Debian 11 (2021).

**Comment by abulhol** [comment]
I just want to comment that we have been running the dev-branch on several servers with Ubuntu 22.04 for half a year, and we see no issues. 
So while that's good enough for us, we would also rather install a release version than a dev-branch which is changing over time.

**Comment by Neustradamus** [comment]
@Sebastian-Roth: Debian 12 "Bookworm" Alpha 2 out:
- https://www.debian.org/devel/debian-installer/
- https://lists.debian.org/debian-boot/2023/02/msg00200.html

"FEATURE_C12" bug:
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1031622
- https://lists.debian.org/debian-boot/2023/02/msg00202.html

**Comment by Sebastian-Roth** [comment]
@Neustradamus I have some good news. There seems to be just one simple minor issue left for FOG to run on Debian 12:
```
$ lsb_release -rs
No LSB modules are available.
n/a
```
We expect to read the version information using `lsb_release` which fails to return the version. Adjusting the script to work around this issue I was able to install FOG dev-branch on Debian 12 no problem. A new FOG release is soon to come.

**Comment by Sebastian-Roth** [comment]
I have been using FOG dev-branch on Debian 12 for testing in the last days and it's working all fine. Closing this issue as fixed as we'll release 1.5.10 pretty soon.

**Comment by Neustradamus** [comment]
@Sebastian-Roth: Thanks a lot!

A lot of new users will arrive with the new OS support :)
